### PR TITLE
New report 'Personal settings > Custom Variables' displays custom variable usage

### PR DESCRIPTION
**Pull request by tsteur**
Shows which custom variable slots are used and which ones aren't. This is useful if you want to get an overview of all used custom variable names and slots and to see which slots are unused. It currently shows all usages, even the one that were used only once or fake once. We could maybe add a parameter later to filter these usages.

There are 2 failing system tests where `idsubdatatable` is increased now: https://travis-ci.org/piwik/piwik/jobs/88684701#L613 Not sure why this is the case and how this is related. I did not just want to update the expected xml. Maybe someone has an idea?

Another system test is failing re Anonymous tracking plugin. We should fix this test once we merge or once it is good to merge. I did not want to add a new expected file to the plugin in case we rename the API method.

_Needed Access_ 
Currently, `Admin` access is required to request API to get usages of slots and to see the "Custom Variables" page but we could maybe as well allow it for `View` access? The `Add new custom variable` button would need to be made only visible for super users then.

_Known issue:_
The API output is incomplete for any aggregated reports (`range`, `year`, ...) since aggregating metadata is not supported yet. It only uses the metadata of the row having the most actions/visits: https://github.com/piwik/piwik/blob/2.15.0/core/DataTable/Row.php#L520

Example: One day you use a certain custom variable in slot 1, the next day in slot 2. It would only show the usage of the slot having more actions or visits. Not both slot 1 and 2.

Ideally it would combine all scopes (unique). Should I work on that or is it alright?

### CODE REVIEW

File: plugins/CustomVariables/lang/en.json [hunk 1]
```diff
@@ -6,6 +6,16 @@
         "CustomVariablesReportDocumentation": "This report contains information about your Custom Variables. Click on a variable name to see the distribution of the values. %s For more information about Custom Variables in general, read the %sCustom Variables documentation on piwik.org%s",
         "PluginDescription": "Custom Variables are (name, value) pairs that you can assign using the Javascript API to visitors or any of their action. Piwik will then report how many visits, pages, conversions for each of these custom names and values. View the detailed Custom Variables for each user and action in the Visitor Log.<br />Required to use <a href=\"http://piwik.org/docs/ecommerce-analytics/\">Ecommerce Analytics</a> feature!",
         "ScopePage": "scope page",
-        "ScopeVisit": "scope visit"
+        "ScopeVisit": "scope visit",
+        "ManageDescription": "This overview shows all custom variable slots and their usages. The names within each slot are ordered by how often they were used in total.",
```
Comment (mattab) [inline plugins/CustomVariables/lang/en.json]: Could you add the name of the website so it is clear that this report depends on website selected eg. `This overview shows all custom variable slots and their usages for %s.`

**Comment by mattab** [comment]
> Ideally it would combine all scopes (unique). 

:+1:  when showing the values in our public API we should aim to make them correct so users don't get disappointed 

Feedback:

 * simple &useful!
 * Add help icon to link to: http://piwik.org/docs/custom-variables/
 * Move the menu entry below "Goals" and above "Tracking code"
 * Could we add a sentence to explain that this report's data will be populated over time after the upgrade to Piwik 2.15.0?

**Comment by tsteur** [comment]
Done apart from "Could we add a sentence to explain that this report's data will be populated over time after the upgrade to Piwik 2.15.0?"

Not sure about this one as it would be there kinda forever even though data might be populated. We could maybe show it in the help text but there not many people would see it.

**Comment by mattab** [comment]
> Not sure about this one as it would be there kinda forever even though data might be populated. We could maybe show it in the help text but there not many people would see it. 

Could we maybe display the notification when:
 * there is some custom data report (not empty report)
 * and there is no slot information yet 

This way, users will know that UI is not broken but that it will take a day or two to fill in (and a few weeks to be fully accurate)

**Comment by tsteur** [comment]
Done. I would have added a test for it but it's not trivial and not really worth it

**Comment by mattab** [comment]
Looks great! :+1: to merge (couldn't directly merge as there are conflicts)

**Comment by mattab** [comment]
Documented this nice feature in  a small section in the [Custom Variables user guide ](https://piwik.org/docs/custom-variables/#usage-of-your-custom-variables-slots)

**Pull request merged**